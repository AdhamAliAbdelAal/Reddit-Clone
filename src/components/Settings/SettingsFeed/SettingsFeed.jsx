import { Box } from '@mui/system';
import { Switch } from '@mui/material';
import { useState, useEffect } from 'react';
import useFetch from '../../../hooks/useFetch';
import {
  AntSwitch, ContentSubHeader, ContentHeader, Content, Button, Header,
  SettingsPageConranier, SubHeader,
} from '../styles';
import settingsPost from '../server';
/**
 * - SettingsFeed
 * - Change Adult content and  Autoplay media in settings page
 *  @component
 *  @property {function} setPrefs set prefs of user
 *  @property {Object} prefs prefs of user
 *  @property {Object} data data from fetch
 */
function SettingsFeed() {
  const api = '/users/me/prefs';
  const [data, dataError] = useFetch(api);
  const [prefs, setPrefs] = useState();
  useEffect(() => {
    // console.log(data?.settings.prefs);
    setPrefs(data?.prefs);
    console.log(dataError);
  }, [data, dataError]);

  return (
    data === null ? (<div data-testid="settings-feed"> error in fecting</div>)
      : (
        <SettingsPageConranier data-testid="settings-feed">
          <Box sx={{ maxWidth: '688px', flex: '1 1 auto' }}>
            <Header>Feed settings</Header>
            <SubHeader>CONTENT PREFERENCES</SubHeader>
            <Button>
              <Content>
                <ContentHeader>
                  Adult content
                </ContentHeader>
                <ContentSubHeader>
                  Enable to view adult and NSFW (not safe for work) content in your feed and search results.
                </ContentSubHeader>
              </Content>
              <AntSwitch onClick={() => { setPrefs((oldPrefs) => ({ ...oldPrefs, adultContent: !oldPrefs.adultContent })); settingsPost({ ...prefs, adultContent: !prefs?.adultContent }); }}>
                <Switch checked={prefs?.adultContent || false} />
              </AntSwitch>
            </Button>
            <Button>
              <Content>
                <ContentHeader>
                  Autoplay media
                </ContentHeader>
                <ContentSubHeader>
                  Play videos and gifs automatically when in the viewport.
                </ContentSubHeader>
              </Content>
              <AntSwitch onClick={() => { setPrefs((oldPrefs) => ({ ...oldPrefs, autoplayMedia: !oldPrefs.autoplayMedia })); settingsPost({ ...prefs, autoplayMedia: !prefs?.autoplayMedia }); }}>
                <Switch checked={prefs?.autoplayMedia || false} inputProps={{ 'aria-label': 'controlled' }} />

              </AntSwitch>
            </Button>
          </Box>
        </SettingsPageConranier>
      )
  );
}

export default SettingsFeed;
