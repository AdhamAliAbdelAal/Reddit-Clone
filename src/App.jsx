import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { CssBaseline, ThemeProvider } from '@mui/material';

import MainNavBar from './components/MainNavBar/MainNavBar';
import SettingsProfile from './components/Settings/SettingsProfile/SettingsProfile';
import SettingsPrivacy from './components/Settings/SettingsPrivacy/SettingsPrivacy';
import SettingsFeed from './components/Settings/SettingsFeed/SettingsFeed';
import SettingsAccount from './components/Settings/SettingsAccount/SettingsAccount';
import Settings from './pages/Settings';

import SignUpPage from './pages/SignUpPage';
import LogInPage from './pages/LogInPage';
import ForgetUsernamePage from './pages/ForgetUsernamePage';
import ForgetPasswordPage from './pages/ForgetPasswordPage';
import ResetPasswordPage from './pages/ResetPasswordPage';

import HomePage from './pages/HomePage';
import Messages from './pages/Messages';
import Moderation from './pages/Moderation';
import Profile from './pages/Profile';
import Search from './pages/Search';
import theme from './styles/theme';
import CreatePost from './pages/CreatePost';
import PostTypeContextProvider from './contexts/PostTypeContext';
import Notifications from './pages/Notifications';
import SubReddit from './pages/SubReddit';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <PostTypeContextProvider>
        <Router className="App">
          <MainNavBar />
          <Routes>
            <Route
              path="/"
              element={
                <HomePage data-testid="home page" />
          }
            />
            <Route
              path="/:postClass"
              element={
                <HomePage />
          }
            />
            <Route
              path="/user"
              element={
                <Profile />
              }
            />
            <Route
              path="/register"
              element={
                <SignUpPage />
}
            />
            <Route
              path="/user/:username/"
              element={
                <Profile />
          }
            >
              <Route
                path=":subTitle"
                element={
                  <Profile />
          }
              >
                <Route
                  path=":sort"
                  element={
                    <Profile />
          }
                />

              </Route>
            </Route>

            <Route
              path="/login"
              element={
                <LogInPage />
          }
            />
            <Route
              path="/password"
              element={
                <ForgetPasswordPage />
          }
            />
            <Route
              path="/resetpassword/:token"
              element={
                <ResetPasswordPage />
          }
            />

            <Route
              path="/username"
              element={
                <ForgetUsernamePage />
          }
            />
            <Route
              path="/search"
              element={
                <Search />
          }
            />
            <Route
              path="/messages"
              element={
                <Messages />
          }
            />
            <Route
              path="/moderation"
              element={
                <Moderation />
          }
            />
            <Route
              path="/submit"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/submit/:subredditName"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/notifications"
              element={
                <Notifications />
          }
            />
            <Route
              path="/SubReddit/:Name"
              element={
                <SubReddit />
          }
            >
              <Route
                path=":postClass"
                element={
                  <SubReddit />
          }
              />
            </Route>
            <Route
              path="/settings"
              element={
                <Settings />
          }
            >
              <Route
                path=""
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="account"
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="profile"
                element={
                  <SettingsProfile />
          }
              />
              <Route
                path="privacy"
                element={
                  <SettingsPrivacy />
          }
              />
              <Route
                path="feed"
                element={
                  <SettingsFeed />
          }
              />
            </Route>
          </Routes>
        </Router>
      </PostTypeContextProvider>
    </ThemeProvider>
  );
}

export default App;
