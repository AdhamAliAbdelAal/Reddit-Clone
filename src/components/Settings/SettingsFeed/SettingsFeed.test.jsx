import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  render, screen,
} from '@testing-library/react';
import SettingsFeed from './SettingsFeed';
import App from '../../../App';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(
    <Router>
      <SettingsFeed />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
// render settings-privacy page
test('should render settings-feed page', () => {
  window.history.pushState({}, '', '/settings/feed');
  render(<App />);
  expect(screen.getByTestId('settings-feed')).toBeInTheDocument();
});
