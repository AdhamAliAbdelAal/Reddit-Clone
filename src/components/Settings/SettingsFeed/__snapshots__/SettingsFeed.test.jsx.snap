// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  data-testid="settings-feed"
>
   error in fecting
</div>
`;
